src/common/utils/url.ts

```diff
@@ -156,32 +159,35 @@ function isManagedResource(serverUrl: URL | string, inputURL: URL | string) {
     managedResources.some((managedResource) => (parsedURL.pathname.toLowerCase().startsWith(`${server.subpath}${managedResource}/`) || parsedURL.pathname.toLowerCase().startsWith(`/${managedResource}/`))));
 }
 
-function getServer(inputURL: URL | string, teams: Team[], ignoreScheme = false): ServerFromURL | undefined {
+function getView(inputURL: URL | string, teams: TeamWithTabs[], ignoreScheme = false): ServerFromURL | undefined {
     const parsedURL = parseURL(inputURL);
     if (!parsedURL) {
         return undefined;
     }
     let parsedServerUrl;
+    let firstOption;
     let secondOption;
-    for (let i = 0; i < teams.length; i++) {
-        parsedServerUrl = parseURL(teams[i].url);
-        if (!parsedServerUrl) {
-            continue;
-        }
-
-        // check server and subpath matches (without subpath pathname is \ so it always matches)
-        if (equalUrlsWithSubpath(parsedServerUrl, parsedURL, ignoreScheme)) {
-            return {name: teams[i].name, url: parsedServerUrl, index: i};
-        }
-        if (equalUrlsIgnoringSubpath(parsedServerUrl, parsedURL, ignoreScheme)) {
-            // in case the user added something on the path that doesn't really belong to the server
-            // there might be more than one that matches, but we can't differentiate, so last one
-            // is as good as any other in case there is no better match (e.g.: two subpath servers with the same origin)
-            // e.g.: https://community.mattermost.com/core
-            secondOption = {name: teams[i].name, url: parsedServerUrl, index: i};
-        }
-    }
-    return secondOption;
+    teams.forEach((team) => {
+        const srv = new MattermostServer(team.name, team.url);
+        team.tabs.forEach((tab) => {
+            const tabView = getServerView(srv, tab);
+            parsedServerUrl = parseURL(tabView.url);
+            if (parsedServerUrl) {
+                // check server and subpath matches (without subpath pathname is \ so it always matches)
+                if (equalUrlsWithSubpath(parsedServerUrl, parsedURL, ignoreScheme)) {
+                    firstOption = {name: tabView.name, url: parsedServerUrl};
```

Comment:
Is it a problem that we might overwrite firstOption and secondOption since we aren't breaking out of this loop immediately? Like if multiple tabs match the URL, then the second one will overwrite the `firstOption` set for the first tab.

Reply:
Yeah I refactored this because you can't break the loop of a `forEach` and I was asked to change it above. I'm not sure if it makes sense to do this or just go back to the `for` loop.

@Willyfrog I can't think of a case where we would match more than one URL unless we had 2 of the same server configured, which I'm pretty we don't allow. Does this seem fine or should we go back to what we had?

Reply:
I don't think it would be a problem, if we ever get into that situation (unlikely, but sometimes the unexpected happens) it means that both are valid options so I'm not sure which of those is the good one, so one or the other is fine to me.

Anyway it was a suggested change, we can revert back if you foresee any problems, there was nothing wrong with the previous code other than being less legible.

Reply:
Let's leave as-is for now. We have lots of time for this to bake in and be tested.